visualization/app/codeCharta/util/algorithm/treeMapLayout/treeMapGenerator.ts

```diff
@@ -241,17 +248,19 @@ function isOnlyVisibleInComparisonMap(node: CodeMapNode, dynamicSettings: Dynami
 }
 
 // Only exported for testing.
-export function calculateAreaValue(node: CodeMapNode, { dynamicSettings }: State) {
+export function calculateAreaValue(node: CodeMapNode, state: State, maxWidth: number) {
```

Comment:
I would keep the deconstruction of `dynamicSettings`. You can do this wihtin the method signature (parameter list) or within the method body. Example for the former one:
```suggestion
export function calculateAreaValue(node: CodeMapNode, { dynamicSettings, appSettings }: State, maxWidth: number) {
```